src/GafferScene/SetFilter.cpp

```diff
@@ -129,7 +169,12 @@ unsigned SetFilter::computeMatch( const ScenePlug *scene, const Gaffer::Context
 	}
 
 	const ScenePlug::ScenePath &path = context->get<ScenePlug::ScenePath>( ScenePlug::scenePathContextName );
-	ConstPathMatcherDataPtr set = scene->set( setPlug()->getValue() );
+
+	Gaffer::ContextPtr pathMatcherContext = new Gaffer::Context( *context, Gaffer::Context::Borrowed );
```

Comment:
> Let's use the fancy new ScenePlug::GlobalScope here.

We can't, can we? That one not only throws away `scene:path` but also the scene that is needed for creating the `ExpressionAst`?

https://github.com/GafferHQ/gaffer/blob/04fa7b2b729eb18e4012603bfd46c05b19bbb92e/src/GafferScene/ScenePlug.cpp#L287